docs/admin/database.rst

```diff
@@ -23,13 +23,198 @@ Migrating MySQL Data to PostgreSQL
+    The Liquibase migrations assume that they have got their original name which contains the case-sensitive prefix ``FK``.
 
 
-.. note::
-    Start Artemis at least once in version 5.12.9 or greater to make sure the current database schema is PostgreSQL-compatible.
-    Only Artemis 6.0.0 or newer can connect to a PostgreSQL database.
+#. Start Artemis at least once in version ``V`` ≧ 6.0.0 or greater to make sure the current database schema is PostgreSQL-compatible.
 
+#. Stop Artemis.
+
+#. Create a database backup using ``mysqldump``.
+   This dump is called ``Artemis.sql`` in the following steps.
+
+#. Copy the ``docker-compose.yml`` file into the same directory as the ``Artemis.sql`` database dump
+   and run the following commands to convert the ``Artemis.sql`` dump into ``Artemis.pg.sql`` that is usable by PostgreSQL.
+
+   .. code-block:: yaml
+        :caption: ``docker-compose.yml`` with helper containers for MySQL and PostgreSQL.
+
+        ---
+        services:
+            mysql:
+                image: docker.io/library/mysql:8
+                environment:
+                    - MYSQL_ROOT_PASSWORD=cRS2N4X2
+                    - MYSQL_DATABASE=Artemis
+                ports:
+                    - 3306:3306
+                command: >
+                    mysqld
+                        --lower_case_table_names=1 --skip-ssl
+                        --character_set_server=utf8mb4
+                        --collation-server=utf8mb4_unicode_ci
+                        --explicit_defaults_for_timestamp
+                        --default-authentication-plugin=mysql_native_password
+                networks:
+                    - db-migration
+
+            postgres:
+                image: docker.io/library/postgres:15
+                environment:
+                    - POSTGRES_USER=root
+                    - POSTGRES_PASSWORD=cRS2N4X2
+                    - POSTGRES_DB=Artemis
+                ports:
+                    - 5432:5432
+                networks:
+                    - db-migration
+
+        networks:
+            db-migration:
+                driver: "bridge"
+        ...
+
+   .. code-block:: bash
+        :caption: Commands to transform the MySQL dump into a PostgreSQL one.
+
+        #! /usr/bin/env bash
+
+        # start the temporary MySQL and Postgres containers
+        docker compose up -d
+
+        # import database dump into MySQL
+        docker compose exec -T mysql mysql --password=cRS2N4X2 < Artemis.sql
+
+        # use pgloader to transfer data from MySQL to Postgres
+        NETWORK_NAME=$(docker network ls | grep "db-migration" | awk '{ print $2 }')
+        docker run --rm --network="${NETWORK_NAME}" docker.io/dimitri/pgloader pgloader mysql://root:cRS2N4X2@mysql/Artemis postgresql://root:cRS2N4X2@postgres/Artemis
+
+        # dump the Postgres data in a format that can be imported in the actual database
+        docker compose exec -T postgres pg_dump -Ox Artemis > Artemis.pg.sql
```

Comment:
Just tested the PR until here because then I compared the ``.sql`` files and wasn't sure if this is correct as when I searched for ``TK`` some of the fields in the ``pg.sql`` weren't in upper cases as before?!

![Screenshot from 2023-03-24 12-16-50](https://user-images.githubusercontent.com/33753999/227534738-b89cfc66-e192-46b6-a4af-e8c256fda6f7.png)

Reply:
Yes, that is indeed a broken version since the keys changed names.
That’s why the additional steps below are needed to merge the working schema dump with the data dump.

The comment should be changed to _not_ mention that it can be imported into the actual database. That might be a leftover from v1 of the steps.

Reply:
![Screenshot from 2023-03-24 15-00-51](https://user-images.githubusercontent.com/33753999/227541868-814b5230-4c7f-43d0-b4cc-133c390093eb.png)

I'm not totally into the problem. This is the output after the merge. Still not sure if this is correct?!

Reply:
Yes, in this case this seems to be the correct behaviour, since the constraint has the same name in `empty.pg.sql`. So Liquibase handles the correct casing in this case.

One example where pgdump changes names is
```sql
-- correct
CREATE INDEX "FK11jgyics37wyibhhpp866exa8" ON public.quiz_statistic_counter USING btree (drag_and_drop_question_statistic_id);
```
vs
```sql
-- pgdump renamed
CREATE INDEX idx_16888_fk11jgyics37wyibhhpp866exa8 ON artemis.quiz_statistic_counter USING btree (drag_and_drop_question_statistic_id);
```

Reply:
Okay then it seems to be alright in the ``merged.pg.sql``.